controllers/openstackdataplanenode_controller.go

```diff
@@ -382,6 +382,9 @@ func (r *OpenStackDataPlaneNodeReconciler) GetInstanceRole(ctx context.Context,
 			Namespace: instance.Namespace,
 			Name:      instance.Spec.Role,
 		}, instanceRole)
+		if err == nil {
+			err = instance.Validate(*instanceRole)
```

Comment:
I'm lat to this, sorry. So, if there are fields no longer unique between role and nodes, this would lock the controller out from reconciling any changes. Is that what we expect? My understanding is that any validation should happen on the API change submission, by validating/mutating webhooks. Not sure about the need for futher validation of internal state after weebhooks did the job, probably such extra validations in the controller code should only warn users, at the most?

Reply:
we can't do the API validation, because node and role share the same data structure, but I didn't know about the webhooks, could you please share an example of a webhook validation?